web/src/recent_view_ui.ts

```diff
@@ -497,6 +497,90 @@ export function process_messages(
     }
 }
 
+function get_unread_count(data: ConversationData): number {
+    if (data.type === "private") {
+        return unread.num_unread_for_user_ids_string(data.to_user_ids);
+    }
+    return unread.num_unread_for_topic(data.stream_id, data.topic);
+}
+
+function get_display_reply_to(conversation_data: ConversationData): string {
+    assert(conversation_data.type === "private");
+    const user_ids = conversation_data.to_user_ids
+        .split(",")
+        .map((user_id) => Number.parseInt(user_id, 10));
+    return message_store.get_pm_full_names(user_ids);
+}
+
+function get_pm_url(conversation_data: ConversationData): string {
+    assert(conversation_data.type === "private");
+    const user_ids = conversation_data.to_user_ids
+        .split(",")
+        .map((user_id) => Number.parseInt(user_id, 10));
+    const pm_url = people.pm_with_url_from_user_ids(user_ids);
+    assert(pm_url !== undefined);
+    return pm_url;
+}
+
+// This mirrors the structure of `echo.build_display_recipient`, using data
+// found in `ConversationData`.
```

Comment:
It felt like it would be pretty hard to generalize `echo.build_display_recipient` to work for this case too, so I just built a variation on it for here.